kpm/lib/kpm/base_artifact.rb

```diff
@@ -164,8 +164,10 @@ def skip_if_exists(artifact_info, coordinates, sha1_file)
         end
 
         # Finally check if remote_sha1 matches what we have locally
-        local_sha1 = Digest::SHA1.file(artifact_info[:file_path]).hexdigest
-        local_sha1 == artifact_info[:sha1]
+        if !artifact_info[:is_tgz]
+          local_sha1 = Digest::SHA1.file(artifact_info[:file_path]).hexdigest
+          local_sha1 == artifact_info[:sha1]
+        end
```

Comment:
Ok, so it turns out we actually need this :(

For generic tgz artifacts (not plugins), the `artifact_info[:file_path]` is the folder where the artifact will be extracted to (destination path). The destination path should always exist, so this block of code will never return false:
```ruby
if artifact_info[:is_tgz]
  artifact_path = Pathname.new(artifact_info[:file_path])
  if artifact_info[:plugin_name]
    artifact_path = artifact_path.join(artifact_info[:plugin_name]).join(artifact_info[:version])
  end
  return false if !artifact_path.exist?
```
Note that `artifact_info[:plugin_name]` is always null in this case, b/c this is not a plugin.

In the old code, this block would always return false for generic tgz artifacts:
```
if !File.exists?(artifact_info[:file_path]) || File.directory?(artifact_info[:file_path])
  return false
end
```
But that code is not executed for tgz packages anymore.